### DISCUSSION TITLE
Anniversary!

### DISCUSSION THREAD

**Discussion by stsp** [dosemu2/dosemu2#1855] [General]
![Снимок экрана от 2022-12-26 11-49-26](https://user-images.githubusercontent.com/3295627/209514727-a218df4f-b429-4ce3-96e1-fe224bb2ce15.png)

Hi guys!
We all celebrate anniversaries sometimes. :)
For humans it is measured in years, and for
software - in commits. Most software would
use the 10000-fold numbers. But if your software
is an early alpha and you are afraid of overflows
and Y2K bugs, then 10000-1 is a safer choice. :)
On the screen-shot above you can see the
whooping 9,999 commits!
Many thanks to those who contributed!
Well, you can see all of them on the same
screen-shot, in the bottom-right corner.
Of course @bartoldeman and @andrewbird 
deserve the special mention for many useful
contributions, thanks guys. :)

Now if we are at it, its a good chance for
visiting our friends and competitors.
Even if we can't compete with them in
some areas, its quite remarkable to see
that we actually do outnumber some of
them by commits.
Eg dosbox-staging has 9,925 commits.
dosbox-x has 17,440 commits - oh well,
older brother. :)
bochs has 12,781 commits.
So lets say its not like we are seriously
outnumbered by anyone besides
dosbox-x.

Hope the next anniversary will be at a
glorious 20,000 rather than on some shy
19,999. :) And we'll be out of this permanent
"not yet written" stage that actually is
already becoming quite annoying to stay
into. Even now there are already some
vague signs of a possible improvements:
full 64bit awareness and an ARM port
speak for themselves. So the next 10K
commits should not be spent on repeatedly
rewriting the dosemu1 code, as it was
before. Even if some dosemu1 code is
still there to rewrite...

Have a good day!

**Comment by jharrison03** [comment]
Happy anniversary and a big thanks to the very responsive and professional crew steering this ship.

**Comment by jschwartzenberg** [comment]
Congratulations!! According to Wikipedia DOSEMU actually turned 30 this year:
> Initial release | September 3, 1992; 30 years ago

Does anybody know if this is accurate?

**Reply by stsp** [reply]
There is `ChangeLog.ancient` that starts
from `Sat Jan 16 10:28:23 1993`. So no
idea what was in 1992. :)
If you look into the linux history page, then
the first version released under GPL, was
published "In the middle of December 1992".
Which would mean that dosemu was released
before Linus even choose the license. Which
sounds not very realistic, but who knows.

At that time I would be able to hear about
dosemu only be it released for zx-spectrum.
And even in late 90x I was using it mainly
for zx-spectrum emulators. :) But eventually
I've found out that my streamer doesn't
work under dosemu (driver doesn't detect
IRQ), so I had to write patches. Eventually
I've found out that the patch or two can
make a certain game to run a bit longer
before the crash, and so I though "well,
maybe I can fix this all myself, if they
obviously can't?". That was the wrong
kind of question. :) The answer haven't
materialized even 25 years later.

**Reply by jschwartzenberg** [reply]
I think it's realistic it was done Linus' license choice. I also found this archive:
https://www.ibiblio.org/pub/historic-linux/ftp-archives/tsx-11.mit.edu/Oct-07-1996/mail-archive/linux-activists/Volume4/CONTENTS
It mentions DOSEMU 0.4 in a digest from the 12th of January 1993. So some versions probably preceded it.

In the 90s one of the most important benchmarks for any operating system was it's DOS compatibility. There were also not that much GNU/Linux native programs back then, so I suspect DOSEMU was quite relevant early on.

Some old stuff (some files are dated 26th of November 1992):
http://discmaster.textfiles.com/browse/10595/libris_3.zip/libris_3/UTILS/DOS_UTIL/H336C.ZIP/DOSEMU.TPZ/DOSEMU.tar/usr/src/dosemu

All results before 1995:
http://discmaster.textfiles.com/search?q=dosemu&tsMax=19950101

**Reply by jschwartzenberg** [reply]
Another reference for the 3rd of September date:
http://discmaster.textfiles.com/view/8177/Night%20Owl%20(NOPV-14)(Night%20Owl%20Publishing)(1994).ISO/011a/dosemu.zip/DOSEMU.HOW

@macleajb, maybe you have some info? :)

**Reply by bartoldeman** [reply]
September 3, 1992 is correct, I created a historical git archive some years ago here:
https://sourceforge.net/p/dosemu/history/ci/master/tree/

```
commit 2abae5a6d73d369de06e05b5dd59d680683b2d35
Author: Matthias Lautner <lautner@informatik.uni-wuerzburg.de>
Date:   Thu Sep 3 18:48:27 1992 +0200

    92/09/03 ver. 0.1 released
```
so indeed over 30 years old, just a year younger than the Linux kernel (initial release Sep 17 but announcement Aug 25, 1991)

Robert Sanders started the changelog, that's where the 1993 comes from (between 0.4 and 0.47, haven't found any intermediate releases). My first contribution was in May 2000, for vgaemu, also quite some time ago!

**Reply by stsp** [reply]
> September 3, 1992 is correct, I created a historical git archive some years ago here:

Its visible also in dosemu2 if you
put this to your .git/config:
`fetch = +refs/replace/*:refs/replace/*`
But I don't know how to make this
automatic.

**Reply by stsp** [reply]
And in fdpp similar line extends the
history up to `Sat May 6 19:34:20 2000 +0000`.

**Reply by stsp** [reply]
I can even do the dummy merge
commit. But in this case git shows
the history commits not when they
were made, but when Bart committed
them to his historic repo...
So it seems making this automatic
is quite impossible. :(

**Comment by stsp** [comment]
Another anniversary here!
11111 commits.
![изображение](https://github.com/dosemu2/dosemu2/assets/3295627/da1f81f5-d2b1-45ce-a940-62fc92ff2837)

Well, the next one seems to be 22222,
which is not to expect any soon. :)
